04_numeronym/kasunfdo/main_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote(`[a11y K8s abc]
+`)
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestNumeronyms(t *testing.T) {
+	actual := numeronyms("testString", "www", "W3school")
+	expected := []string{"t8g", "www", "W6l"}
+	for i, v := range actual {
+		if expected[i] != v {
+			t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected[i], v)
+		}
+	}
+}
+
+func TestGetNumeronyms(t *testing.T) {
+	tests := map[string]struct {
+		input    string
+		expected string
+	}{
+		"Empty string": {
+			input:    "",
+			expected: "",
+		},
+		"ASCII string": {
+			input:    " Multilingualization",
+			expected: "M17n",
+		},
+		"Rune string": {
+			input:    " ➳↪▲✓₷$€₡😀⤆ ",
+			expected: "➳8⤆",
+		},
+	}
```

Comment:
You should add some additional test cases for the boundary conditions - strings of length 3 and 4 to make sure the conditional logic in the function is correct. I know you have "www" in the earlier test case, but it really belongs here as this is the function with the condition on string length, so this is where the test should be.